src/ol/source/tilewmssource.js

```diff
@@ -277,6 +283,19 @@ ol.source.TileWMS.prototype.getRequestUrl_ =
 
 
 /**
+ * @param {ol.proj.Projection} projection Projection.
+ * @return {ol.tilegrid.TileGrid} Tile grid.
+ */
+ol.source.TileWMS.prototype.getTileGridForProjection = function(projection) {
+  if (goog.isNull(this.tileGrid)) {
+    return ol.tilegrid.getForProjection(projection, this.maxZoom_);
```

Comment:
You should use `ol.tilegrid.createForProjection` here - otherwise a tile grid with a specific maxZoom will be cached on the projection instance.